internal/controller/instaslice_daemonset.go

```diff
@@ -211,96 +200,101 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 						log.FromContext(ctx).Error(shutdownErr, "error to perform nvml.Shutdown")
 					}
 				}()
-
-				availableGpus, ret := nvml.DeviceGetCount()
 				if ret != nvml.SUCCESS {
 					log.FromContext(ctx).Error(ret, "Unable to get device count")
 				}
 				//TODO: any GPU can fail creating CI and GI
 				// if simulator mode is on do not perform NVML calls
 				// TODO: move this logic to a new vendor specific file
-				if _, exists := cachedPreparedMig[allocations.PodName]; !exists {
+				if prep, exists := cachedPreparedMig[allocations.PodUUID]; !exists || isPreparedMigEmpty(prep) {
 					placement := nvml.GpuInstancePlacement{}
-					for i := 0; i < availableGpus; i++ {
+					device, retCodeForDevice := nvml.DeviceGetHandleByUUID(allocations.GPUUUID)
+					if retCodeForDevice != nvml.SUCCESS {
+						log.FromContext(ctx).Error(ret, "error getting GPU device handle")
```

Comment:
I think device will not be initialized here, missing return?

Reply:
GPU fault tolerance is not handled, decided to log errors and let it complain as node is faulty.

Reply:
This error should not happen if the GPU is healthy. the device fault tolerance needs to be discussed. One option here is to disable InstaSlice CR when we see such errors instead of requeue or retry.

Reply:
comment about it in the code.